### ISSUE TITLE
Is there a v5/v6 problem in MapController?

### ISSUE THREAD

**Issue by Jamie5** [visgl/react-map-gl#1712]
#### Description

In https://github.com/visgl/react-map-gl/blame/5.3-release/src/utils/map-controller.js , we see in line 211 that we do `      this.mapState = prevOptions;`. Puzzlingly, elsewhere `this.mapState` seems to be a `MapState` class, unlike here where `prevOptions` is `this.mapStateProps || {}`. I wonder if this is the intended behavior?

(I understand that the library is moving towards v7, but in the meantime if this is a clear issue with an obvious fix, perhaps it can be fixed?)

#### Repro Steps

Unclear - the error that came in was in a subclass of `MapController`, on the line `const newMapState = this.mapState.zoom({ pos, scale });`, and the reported error said that `e.mapState.zoom is not a function`. It is unclear how this situation arose, and this code pointed out above is not definitely the cause, but it's the only thing suspicious that I can find so far.

#### Environment (please complete the following information):

N/A

#### Logs

N/A

**Comment by Pessimistress** [comment]
I do not believe `mapState` is a documented API. You should always call `getMapState` before attempting to use it.

**Comment by Jamie5** [comment]
Thanks, will try that. The typescript bindings don't mark `mapState` as private but I guess it's all going away in v7 anyways.
